drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/docks/KogitoEditorDock.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.kogito.client.docks;
+
+import java.util.Collection;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.google.gwt.core.client.GWT;
+import org.jboss.errai.ioc.client.api.ManagedInstance;
+import org.kie.workbench.common.widgets.client.docks.AuthoringEditorDock;
+import org.kie.workbench.common.widgets.client.docks.WorkbenchDocksHandler;
+import org.uberfire.client.workbench.docks.UberfireDock;
+import org.uberfire.client.workbench.docks.UberfireDockPosition;
+import org.uberfire.client.workbench.docks.UberfireDocks;
+import org.uberfire.mvp.Command;
+import org.uberfire.mvp.PlaceRequest;
+
+@ApplicationScoped
+public class KogitoEditorDock implements AuthoringEditorDock {
```

Comment:
I think this class should be moved to `kie-wb-common-kogito`. Wdyt? It is a generic implementation of `Dock` from what I see

Reply:
@danielezonca It's seems reasonable, considering `AuthoringWorkbenchDocks` is in the same repo. I'll do it, in a separate ticket.